### PR TITLE
feat(core): CW-21046 refactor apdu command

### PR DESCRIPTION
**Pull request by hankliu418**
----

*Checklist:*

- README.md includes:
  - [ ] The details of signing data and transaction data, this includes parameters.
  - [ ] Official docs or white paper.
  - [ ] Website or api can query assets and broadcast transaction.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by cbx-tech** [comment]
Task linked: [CW-21046 改 core 移除顯示的程式碼](https://app.clickup.com/t/25577573/CW-21046)

**Review by tp26610 — DISMISSED** [review]

**Review by ericting2000 — APPROVED** [review]

File: packages/core/src/apdu/informational.ts [hunk 1]
```diff
@@ -69,6 +70,7 @@ export const getCardInfo = async (
 
 /**
  * Update last used keyId to store in CWS.
+ * @deprecated Please use setting.card.updateKeyId instead
  * @param {Transport} transport
  * @param {Array<{KeyId: string, coinType: string}>} dataArr
  * @param {string} P1
```
Comment (gundomlegend) [inline packages/core/src/apdu/informational.ts:76]: 這邊似乎少 appId、appPrivKey 的敘述

File: packages/core/src/apdu/informational.ts [hunk 2]
```diff
@@ -136,7 +139,7 @@ export const getLastKeyId = async (transport: Transport, P1: string) => {
 };
 
 /**
- *
+ * @deprecated Please use setting.card.toggleDisplayAddress instead
  * @param {Transport} transport
  * @param {string} signature
```
Comment (gundomlegend) [inline packages/core/src/apdu/informational.ts:144]: 好奇 signature 指的是 appId 還是 appPrivKey？

File: packages/core/src/apdu/mcu/dfu.ts [hunk 3]
```diff
@@ -55,6 +55,11 @@ const getMCUInfo = async (transport: Transport): Promise<MCUInfo> => {
   return { hardwareVersion, firmwareVersion, battery };
 };
 
+/**
+ * @deprecated Please use mcu.dfu.checkUpdate instead
+ * @param transport
```
[the remark this answers is not in the document]
Comment (gundomlegend) [reply packages/core/src/apdu/mcu/dfu.ts:60]: 不過 code 先進沒差～註記一下這邊都沒標型別～

File: packages/core/src/wallet/secret.ts [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+  }
+}
+
+/**
+ *
+ * @param {Transport} transport
+ * @param {number} strength
+ */
+export async function initSecureRecovery(transport: Transport, strength: number) {
+  if (transport.cardType === 'Lite') {
```
Comment (Dorac) [inline packages/core/src/wallet/secret.ts:167]: 要不要考慮宣告Enum提高可讀跟維護性?
```
enum CardType {
  Pro = 'Pro',
  Lite = 'Lite'
}
```
介面可以這樣給預設值
```
abstract class Transport {
  ...
  cardType: CardType = CardType.Pro;
}
```

File: packages/core/src/transaction/command.ts [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+ * @param argument smart contract data
+ * @returns encryptedSignature
+ */
+export const executeSegmentScript = async (
+  transport: Transport,
+  appId: string,
+  appPrivKey: string,
+  argument: string
+) => {
+  if (argument.length > 2147483648 * 2) throw new Error('argument too long');
```
Comment (Dorac) [inline packages/core/src/transaction/command.ts:77]: 要不要宣告為常數比較好理解?
const MAX_ARGUMENT_LENGTH = 2147483648; // 2GB

File: packages/core/src/transaction/command.ts [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+ * @param {*} transport
+ * @param {*} argument
+ */
+export const executeUtxoSegmentScript = async (
+  transport: Transport,
+  appId: string,
+  appPrivKey: string,
+  utxoArgument: string
+) => {
+  if (utxoArgument.length > 2147483648 * 2) throw new Error('argument too long');
```
Comment (Dorac) [inline packages/core/src/transaction/command.ts:174]: 同上

File: packages/core/src/info/index.ts [hunk 7]
```diff
@@ -86,4 +87,61 @@ const getMCUInfo = async (transport: Transport): Promise<MCUInfo> => {
...
+    let bipEd25519IsInit;
+    if (databuf.length >= 11) {
+      bipEd25519IsInit = databuf.slice(10, 11).toString('hex');
+    }
+    let accountDigest20;
+    if (databuf.length > 11) {
+      accountDigest20 = databuf.slice(11, 31).toString('hex');
+    }
+
+    if (accountDigest === '81c69f2d90' || accountDigest === '3d84ba58bf' || accountDigest === '83ccf4aab1') {
```
Comment (Dorac) [inline packages/core/src/info/index.ts:124]: 還有印象這個判斷是用來幹嘛的嗎?

**Review by Dorac — APPROVED** [review]

**Review by a0979470582 — APPROVED** [review]

File: packages/core/src/apdu/execute/execute.ts [hunk 8]
```diff
@@ -111,6 +112,9 @@ export const executeCommand = async (
   params2?: string
   // forceUseSC: boolean = false,
 ): Promise<{ statusCode: string; msg: string; outputData: string }> => {
+  if (transport.cardType === 'Lite' && executedTarget === target.MCU) {
```
Comment (a0979470582) [inline packages/core/src/apdu/execute/execute.ts:115]:
```suggestion
  // import { CardType } from '.....';
  // ....
  if (transport.cardType === CardType.Lite && executedTarget === target.MCU) {
```

File: packages/core/src/apdu/mcu/display.ts [hunk 9]
```diff
@@ -57,6 +67,9 @@ export const updateBalance = async (
   appPrivKey: string,
   data: Array<{ balance: number; coinType: string }>
 ) => {
+  if (transport.cardType === 'Lite') {
```
Comment (a0979470582) [inline packages/core/src/apdu/mcu/display.ts:70]: 這檔案有三個 'Lite' 也可換 CardType.Lite

**Pull request merged**